### PR TITLE
DEV: Add custom error messages to form template forms

### PR DESCRIPTION
**Pull request by keegangeorge**
This PR adds custom error message strings to form template forms validation messages. This allows for less flakiness in specs and also assists in easy localizations.

### CODE REVIEW

**Comment by keegangeorge** [comment]
Failing tests seem to be unrelated?

File: config/locales/client.en.yml [hunk 1]
```diff
@@ -4507,6 +4507,30 @@ en:
...
+          select-multiple: "Please select at least one item in the list."
+          checkbox: "Please check this box if you want to proceed."
+        typeMismatch:
+          default: "Please enter a valid value."
+          color: "Please enter a color."
+          date: "Please enter a date."
+          email: "Please enter a valid email address."
+          number: "Please enter a number."
+          password: "Please enter a valid password."
+          tel: "Please enter a valid telephone number"
```
Comment (pmusaraj) [inline config/locales/client.en.yml:4525]:
```suggestion
          tel: "Please enter a valid telephone number."
```

For consistency.

**Review by pmusaraj — APPROVED** [review]

**Comment by pmusaraj** [comment]
> Failing tests seem to be unrelated?

Yes, maybe run the footnote tests locally just to be sure?

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -4507,6 +4507,30 @@ en:
...
+          email: "Please enter a valid email address."
+          number: "Please enter a number."
+          password: "Please enter a valid password."
+          tel: "Please enter a valid telephone number."
+          text: "Please enter a text value."
+          url: "Please enter a valid URL."
+        tooShort: "The input must be %{minLength} characters or longer."
+        tooLong: "The input must be less than %{maxLength} characters."
+        rangeOverflow: "The input must be less than %{max}."
+        rangeUnderflow: "The input must be more than %{min}."
```
Comment (gschlager) [inline config/locales/client.en.yml:4531]: These string need to be pluralized. For example:

```
tooShort:
  one: "The input must be %{count} character or longer."
  other: "The input must be %{count} characters or longer."
```

See `/t/translation-guidelines-for-developers-and-designers/91654#numbers-plurals-6`

cc @pmusaraj

**Pull request merged**